### DISCUSSION TITLE
ddev pull pantheon fails to pull files because invalid asset

### DISCUSSION THREAD

**Discussion by Bob-E** [ddev/ddev#5296] [General]
`ddev pull pantheon` results in `Pull failed: Unable to validate import asset /Users/username/Sites/sitename/.ddev/.downloads/files: invalid asset: provided path is not a .sql file or archive`

`ddev pull pantheon` works fine on another machine, just not this one I've just set up recently!

**Comment by rfay** [comment]
Hi @Bob-E - Could you please provide the output of `ddev debug test` as a gist on gist.github.com (or any pastebin, or as a file)?

Does `ddev pull pantheon --skip-files` work?
Does `ddev pull pantheon --skip-db` work?
Have you edited the pantheon.yaml? Could you provide it with any secrets hidden?

**Reply by Bob-E** [reply]
both `--skip-files` and `--skip-db` fail

**Reply by Bob-E** [reply]
https://gist.github.com/Bob-E/f3de62b834a6d808e0257a55ef71dbc9

**Reply by Bob-E** [reply]
Thank you for your wisdom!

**Reply by rfay** [reply]
Thanks - please add the .ddev/providers/pantheon.yaml to the gist.  Please confirm that it's exactly the same as the version that works on your other computer.

**Reply by Bob-E** [reply]
Here's the pantheon.yaml
It's currently trying to pull from the live site after several failed attempts to pull from dev.

https://gist.github.com/Bob-E/80f0cd1e22d2e13aff036f68535bc7cf

**Reply by Bob-E** [reply]
I'm working on getting the other pantheon.yaml file :)

**Reply by Bob-E** [reply]
successful ddev pull pantheon from other machine with identical pantheon.yaml file

https://gist.github.com/Bob-E/2dab74397260ba4e4871f461b4c2be46

**Reply by rfay** [reply]
You haven't yet shown me the failed `ddev pull` from the bad machine have you?

I assume you have the same ddev version and the same basic type of mac? Same architecture? Both Intel?

**Reply by Bob-E** [reply]
Both Macs are Intel i7.
The machine that successfuly pulls had `ddev 1.21.4`
and the newer machine that fails is `ddev 1.22.1`

Here is the failed attempt

```
etloaner@UTO-C02D6BPBMD6M sdoc-col % ddev pull pantheon -y
Authenticating...
 [notice] Logged in via machine token.
Obtaining databases...
+ set -eu -o pipefail
+ ls /var/www/html/.ddev
+ pushd /var/www/html/.ddev/.downloads
+ terminus backup:get sdoc-col.live --element=db --to=db.sql.gz
 [notice] Downloading sdoc-col_live_2023-08-24T14-00-00_UTC_database.sql.gz to db.sql.gz
Importing databases [/Users/etloaner/Sites/sdoc-col/.ddev/.downloads/db.sql.gz /Users/etloaner/Sites/sdoc-col/.ddev/.downloads/files]

 839MiB 0:00:50 [16.6MiB/s] [================================>] 100%
Pull failed: Unable to validate import asset /Users/etloaner/Sites/sdoc-col/.ddev/.downloads/files: invalid asset: provided path is not a .sql file or archive
```

**Reply by rfay** [reply]
They are definitely not doing the same thing, or maybe against the same site?. The working one does this:

```
+ terminus backup:get sdoc-col.live --element=db --to=db.sql.gz
 [notice] Downloading sdoc-col_live_2023-08-24T14-00-00_UTC_database.sql.gz to db.sql.gz
Importing databases [/Users/robert/Sites/sdoc-rfi/.ddev/.downloads/db.sql.gz]
```

The failing one does this:
```
+ terminus backup:get sdoc-col.live --element=db --to=db.sql.gz
 [notice] Downloading sdoc-col_live_2023-08-24T14-00-00_UTC_database.sql.gz to db.sql.gz
Importing databases [/Users/etloaner/Sites/sdoc-col/.ddev/.downloads/db.sql.gz /Users/etloaner/Sites/sdoc-col/.ddev/.downloads/files]
```

You'll note that it's trying to import two things when importing the database, and that's just wrong. I think it may have found something unexpected in the terminus backup:get. Please see if the TERMINUS_MACHINE_TOKEN is set to the same thing on each project. 

And of course, you can try the older version of DDEV to see if that's the difference. See https://ddev.readthedocs.io/en/latest/users/usage/faq/#how-can-i-install-a-specific-version-of-ddev - there's no risk to it. Just `brew unlink ddev` assuming you've used brew to install it, and use the script to install the older version.

**Reply by rfay** [reply]
You may want to take a look and delete your .ddev/.downloads directory - maybe there's something stuck in there?

**Reply by Bob-E** [reply]
I will give those all a try. Thank you so much for your help with this, Randy! I'm just trying to onboard at my new job and set up my new machine to work like my personal machine.

**Reply by Bob-E** [reply]
Downgrading to `ddev 1.21.4` did the trick! Thanks again for all your help!

**Reply by rfay** [reply]
Well, I'm glad you did that... but we haven't been having any trouble with automated tests or manual testing of this on v1.22.1.  I hope you'll open an issue if there's actually a problem with v1.22.1

**Reply by Bob-E** [reply]
I will see what happens when I set up ddev 1.22.1 on my new work machine in a couple of weeks. I'll let you know if I get a similar outcome, or, hopefully, it will just work as expected!
